src/opencl/openclwrapper.cpp

```diff
@@ -186,12 +186,12 @@ static ds_status initDSProfile(ds_profile **p, const char *version) {
   }
 
   if (numDevices > 0) {
-    devices.reserve(numDevices);
+    devices.resize(numDevices);
   }
 
   profile->numDevices = numDevices + 1; // +1 to numDevices to include the native CPU
-  profile->devices.reserve(profile->numDevices);
-  memset(&profile->devices[0], 0, profile->numDevices * sizeof(ds_device));
+  profile->devices.resize(profile->numDevices);
+  memset(profile->devices.data(), 0, profile->numDevices * sizeof(ds_device));
```

Comment:
Is the `memset` still needed? The `resize` method should already fill all new elements with zero.

Reply:
You are right, it initializes it with default. My intent was to minimize changes and follow safe programming (e.g. who said that it wouldn't have any values inserted before). So may adress it here (in that PR) or suggest a small refactoring of that function (as for me it has a mix of c/c++ and unndeded complexity) done by me in next PR.

Reply:
Do you plan to use Tesseract with OpenCL or are you already using it? In my tests (done some years ago) it did not work good enough to justify more efforts to improve that code.

Reply:
Instead of modifying line 194 I suggest to remove it in the PR here. Additional changes can be done in another PR later.

Reply:
Doing some PoCs to see if I can use it.